### DISCUSSION TITLE
Sleep timer at chapter end

### DISCUSSION THREAD

**Discussion by Jimmyd1999** [PaulWoitaschek/Voice#2177] [Ideas]
I'm really enjoying using voice app - thanks for building it 

A feature I find useful in audible is to set the sleep timer to stop playback when it reaches the current chapter end, as working out the remaining playback time is not always easy if I'm sleepy, or out running and don't want to get through my book too fast whilst it's difficult to use my phone to stop playback/work out the remaining minutes so I don't creep into the next chapter.

**Comment by samosa-chaat** [comment]
I am also waiting for this feature ...must needed

**Comment by jeremyf** [comment]
I was thinking about this feature, and realized that I can use the Sleep with customer timer that I can set to roughly match the chapter.  So it's adequate but having an option that is "until the end of the chapter" would be wonderful.

**Comment by xgpt** [comment]
I need the one click "at end of chapter" sleep function as I often fall asleep listening to audiobooks and enjoy knowing I haven't blown past a chapter marker.

**Comment by teabag12** [comment]
I am thinking of trying to implement it myself. Being the opposite of an expert, though, some initial advice where to start would be helpful. :)

**Comment by voidzone** [comment]
Sleep at end of chapter vs sleep at end of track/file aren't necessarily the same thing, and likely wildly different difficulties to implement.

Personally, I'd like a sleep at end of track option.

**Comment by yanneves** [comment]
Came here with the same feature in mind and looks like it's already close to being implemented in https://github.com/PaulWoitaschek/Voice/pull/2838
